gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -60,10 +61,60 @@ async fn gatewayd_shows_info_about_all_connected_federations() {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn gatewayd_shows_balance_for_any_connected_federation() -> anyhow::Result<()> {
-    // todo: implement test case
+async fn gatewayd_shows_balance_for_any_connected_federation() {
+    let (_, rpc, fed1, fed2, bitcoin) = fixtures::fixtures().await;
 
-    Ok(())
+    let id1 = fed1.connection_code().id;
+    let id2 = fed2.connection_code().id;
+    let client1 = fed1.new_client().await;
+
+    connect_federations(&rpc, &[fed1, fed2]).await.unwrap();
+
+    let bal1 = rpc
+        .get_balance(BalancePayload { federation_id: id1 })
+        .await
+        .unwrap()
+        .msats;
+
+    let bal2 = rpc
+        .get_balance(BalancePayload { federation_id: id2 })
+        .await
+        .unwrap()
+        .msats;
+
+    assert_eq!(bal1, 0);
+    assert_eq!(bal2, 0);
+
+    let adr1 = rpc
+        .get_deposit_address(DepositAddressPayload { federation_id: id1 })
+        .await
+        .unwrap();
+
+    let adr2 = rpc
+        .get_deposit_address(DepositAddressPayload { federation_id: id2 })
+        .await
+        .unwrap();
+
+    bitcoin.prepare_funding_wallet().await;
+    bitcoin
+        .send_and_mine_block(&adr1, bitcoin::Amount::from_sat(5_000_000))
+        .await;
```

Comment:
As mentioned above, you don't need to do this. Just print the money using the dummy module then verify that `info` returns the expected value.

Reply:
I was using the fed client directly to do this, and still had the issue. I just pushed two commits to make it explicit how the test fails, appreciate a look.